wgsl/index.bs

```diff
@@ -3060,15 +3091,87 @@ Issue:
 #### Input-output Locations TODO #### {#input-output-locations}
 TODO: *Stub*. Location-sizing of types, non-overlap among variables referenced within an entry point static call tree.
 
-### Resource interface TODO ### {#resource-interface}
+### Resource interface ### {#resource-interface}
+
+A <dfn noexport>resource</dfn> is an object,
+other than a [[#pipeline-inputs-outputs|pipeline input or output]],
+which provides access to data external to a [=shader stage=].
+Resources are shared by all invocations of the shader.
+
+There are four kinds of resources:
+
+* [=uniform buffers=]
+* [=storage buffers=]
+* textures
```

Comment:
speaking of consistency, why are we separating the uniform/storage buffers into different resource, but not separating the sampled/storage textures?